src/iqm/iqm_client/transpile.py

```diff
@@ -0,0 +1,392 @@
+# Copyright 2021-2023 IQM client developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Collection of transpile functions needed for transpiling to specific devices
+"""
+from enum import Enum
+from typing import Iterable, Optional
+import warnings
+
+from iqm.iqm_client import Circuit, CircuitExecutionError, Instruction, IQMClient, QuantumArchitectureSpecification
+
+
+class ExistingMoveHandlingOptions(Enum):
+    """Transpile options for handling of existing Move instructions.
+
+    KEEP: the transpiler will keep the move instructions as specified checking if they are correct,
+            adding more as needed.
+    REMOVE: the transpiler will remove the instructions and add new ones.
+    TRUST: the transpiler will keep the move instructions without checking if they are correct,
+            and add more as needed.
+    """
+
+    KEEP = 'keep'
+    REMOVE = 'remove'
+    TRUST = 'trust'
+
+
+class ResonatorStateTracker:
+    """Class for tracking the location of the resonator states on the quantum computer as they are moved with the move
+    gates.
+
+    Args:
+        move_calibrations (dict[str, Iterable[str]]): A dictionary describing between which qubits a move gate is
+        available, for each resonator, i.e. `move_calibrations[resonator] = [qubit]`
+    """
+
+    move_gate = 'move'
+
+    def __init__(self, move_calibrations: dict[str, list[str]]) -> None:
+        self.move_calibrations = move_calibrations
+        self.res_qb_map = {r: r for r in self.resonators}
+
+    @staticmethod
+    def from_quantum_architecture_specification(arch: QuantumArchitectureSpecification):
+        """Constructor to make the ResonatorStateTracker from a QuantumArchitectureSpecification.
+
+        Args:
+            arch (QuantumArchitectureSpecification): The architecture to track the resonator state on.
```

Comment:
```suggestion
    def from_quantum_architecture_specification(arch: QuantumArchitectureSpecification) -> ResonatorStateTracker:
        """Constructor to make the ResonatorStateTracker from a QuantumArchitectureSpecification.

        Args:
            arch: The architecture to track the resonator state on.
```
In other methods also.